jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/service/RegistrationService.java

```
@@ -306,12 +307,20 @@ private boolean validateAccessCode(String accessCode) {
         return accessCode.equals(storedAccessCode);
     }
 
-    private void doUnregister(String eMail, EiamClientRole role) throws UserNotFoundException {
-        Optional<UserInfo> userInfoByMail = userInfoRepository.findByEMail(eMail);
-        UserInfo userInfo = userInfoByMail.get();
+    private void doUnregister(String searchParam, SearchParamType searchParamType, EiamClientRole role) throws UserNotFoundException {
+        UserInfo userInfo = findUserInfo(searchParam, searchParamType);
         userInfo.unregister();
         this.eiamAdminService.removeRole(userInfo.getUserExternalId(), role);
-        LOGGER.info("JobSeeker with email {} has been unregistered", eMail);
+        LOGGER.info("JobSeeker with {} : {} has been unregistered", searchParamType, searchParam);
         applicationEventPublisher.publishEvent(BusinessLogEvent.of(USER_UNREGISTERED).withObjectId(userInfo.getId().getValue()).withAdditionalData("affectedRole", role.getName()));
     }
+
+    private UserInfo findUserInfo(String searchParam, SearchParamType searchParamType) {
+        if (searchParamType == SearchParamType.STES_NR) {
+            return userInfoRepository.findByPersonNumber(Long.valueOf(searchParam)).orElse(null);
+        } else if (searchParamType == SearchParamType.EMAIL) {
+            return userInfoRepository.findByEMail(searchParam).orElse(null);
+        }
+        return null;
```

Please throw an IllegalArgumentException here instead of returning null.